app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt

```diff
@@ -75,11 +75,35 @@ class PGPSettings(
         titleRes = R.string.pref_passphrase_cache_auto_clear_title
         summaryRes = R.string.pref_passphrase_cache_auto_clear_summary
         defaultValue = true
-        /* clear cache once when unchecking; this is to prevent a malicious user
-         * from bypassing cache clearing via the settings */
+        /* Clear the cache once when unchecking; this is to prevent a malicious user (someone
+         * knowing the screen-lock pin, but not knowing the PGP passphrase) from bypassing cache
+         * clearing via the settings. However, clearing EncryptedSharedPreferences requires
+         * authentication, otherwise the app crashes. Thus, the bad user could still bypass cache
+         * clearing by dismissing the auhentication dialog. To prevent this, we enforce cache
+         * clearing to stay enabled in case of any authentication failure. */
         onCheckedChange { checked ->
-          if (!checked)
-            activity.lifecycleScope.launch { passphraseCache.clearAllCachedPassphrases(activity) }
+          if (!checked) {
+            if (BiometricAuthenticator.canAuthenticate(activity)) {
+              BiometricAuthenticator.authenticate(
+                activity,
+                R.string.pref_passphrase_cache_auto_clear_authenticate_disable,
+              ) {
+                if (it is BiometricAuthenticator.Result.Success) {
+                  activity.lifecycleScope.launch {
+                    passphraseCache.clearAllCachedPassphrases(activity)
+                  }
+                } else {
+                  activity.sharedPrefs.edit {
+                    activity.sharedPrefs.edit { remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE) }
+                  }
+                }
+              }
+            } else {
+              activity.sharedPrefs.edit {
+                activity.sharedPrefs.edit { remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE) }
+              }
+            }
```

Comment:
Nesting two calls to `edit` is redundant.

You might need to reformat the code after applying this.

```suggestion
                  activity.sharedPrefs.edit {
                    remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE)
                  }
                }
              }
            } else {
              activity.sharedPrefs.edit {
                remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE)
              }
            }
```

Reply:
Hm, strange. In my latest local version that I force-pushed, this nesting is not present, but on GitHub it indeed looks as you noticed. Maybe I should delete the branch on GitHub and push a corrected version.

Reply:
Will fix it in a moment.

Reply:
Should be ok now.